### PR TITLE
Adds the Cleaving Saw as what will be the reward for killing a miniboss

### PR DESCRIPTION
**Pull request by ChangelingRain**
It's a two-mode weapon that does bonus damage to mining mobs and megafauna.
It is not currently added, but will be a reward for killing a "miner" using it.

In the 'closed' mode it attacks twice as fast as normal melee weapons and applies a 'bleed' on mining mobs and megafauna that will do massive damage if the bleed meter fills up. It's a visible effect.
In the 'open' mode it attacks in a wide swipe, hitting all upright mobs near the target, and does bonus damage to mining mobs and megafauna.
Both modes will build up the bleed, but it's obviously harder to trigger if open due to lowered speed.

- [x] Item and on-mob sprites for it. It's [this thing](http://bloodborne.wiki.fextralife.com/Saw+Cleaver).
- [x] Balance? It should be fine as-is but make sure of this.

### CODE REVIEW

**Comment by DoctoreMad** [comment]
I was about to scream "you can't add a reward for killing other miners because of the fucking freeze" but then I read up the code.
So, huhhh, its adminspawn only for now? That "miner miniboss" isn't coded after all.

**Comment by ChangelingRain** [comment]
Yeah.

**Comment by Kromgar** [comment]
Please make him a giantdad

**Comment by Repukan** [comment]
Muh dick

**Comment by FantasticFwoosh** [comment]
Its fathers day. 

giantdad confirmed.

**Comment by Durkel** [comment]
>miner boss.

You can't add a player reference to ergosavi, you should know better than that.

File: code/datums/status_effects/debuffs.dm [hunk 1]
```diff
@@ -185,3 +185,75 @@
...
+	else
+		qdel(src)
+
+/datum/status_effect/saw_bleed/on_remove()
+	if(needs_to_bleed)
+		var/turf/T = get_turf(owner)
+		new /obj/effect/temp_visual/bleed/explode(T)
+		for(var/d in GLOB.alldirs)
+			new /obj/effect/temp_visual/dir_setting/bloodsplatter(T, d)
+		playsound(T, "desceration", 100, 1, -1)
```
Comment (Xhuis) [inline code/datums/status_effects/debuffs.dm]: I don't know _WHO_ spelled "desecration" like this but I have WORDS TO EXCHANGE

Comment (DoctoreMad) [inline code/datums/status_effects/debuffs.dm]: You mean like, fixing c to be behind the e? That's not a very big exchange of words.

Comment (Xhuis) [inline code/datums/status_effects/debuffs.dm]: Yeah, but it makes me _A N G E R Y_

**Comment by Repukan** [comment]
Why not add it as the big legion (not the giant skull, the beefed up legion that spaws legions) drop to fine tune both the drop and the creature at the same time?

**Comment by ChangelingRain** [comment]
Because legion dropping it is not thematic and not what I'd do for legion. It's not good enough of a reward for that, either.

**Comment by Repukan** [comment]
Not the giant skull one, the beefed up one.

Maybe that great miner fell only when a bunch of legions ganged up on him.

File: code/game/objects/items/weapons/melee/transforming.dm [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		throw_speed = initial(throw_speed)
+		if(attack_verb_off.len)
+			attack_verb = attack_verb_off
+		icon_state = initial(icon_state)
+		w_class = initial(w_class)
+	transform_messages(user, supress_message_text)
+	add_fingerprint(user)
+	return TRUE
+
+/obj/item/weapon/melee/transforming/proc/transform_messages(mob/living/user, supress_message_text)
```
Comment (lzimann) [inline code/game/objects/items/weapons/melee/transforming.dm]: This whole proc can be just
```
playsound(...)
if(!supress_message_text)
    to_chat(user, "<span class='notice'>[src] [active ? "is now active" : "can now be concealed"].</span>")
```

Comment (lzimann) [inline code/game/objects/items/weapons/melee/transforming.dm]: Or if you really don't want a longer `to_chat` message, you can do the same way you did with the cleaving saw's transform messages.

**Comment by WJohn** [comment]
only took me ten trillion years to sprite these things because i'm a lazy pos

File: code/game/objects/items/weapons/melee/transforming.dm (hunk 2 above)
Comment (ChangelingRain) [inline code/game/objects/items/weapons/melee/transforming.dm]: I did this fyi

**Pull request merged**